### DISCUSSION TITLE
API - getPersonAssets - more than 1000 reccords

### DISCUSSION THREAD

**Discussion by Rick45** [immich-app/immich#7534] [General]
Hello everyone, before anything else, great job on the app. It's amazing. 

One thing that I'm miss from google photos is the "smart" albums, where I used to have an album with my wife where it automatically add the pictures of our daughter. 

as that is currently unavailable, I have built a Python script that runs every day at 1AM that queries the person's assets(in this case, my daughter) and adds them to a shared album.

the main issue that I have is that currently, the endpoint getPersonAssets only returns a maximum of 1000 items, and I don't see a way to do any type of skip/paging, so, when I do the initial run, as I have around 1.5k assets for her, the album only receives the latest 1000.

Is there any way to do a paging query that I'm missing?

The endpoint that I'm currently using for grabbing the person assets: https://immich.app/docs/api/get-person-assets

**Comment by michelheusschen** [comment]
This endpoint doesn't support pagination. You can use [searchMetadata](https://immich.app/docs/api/search-metadata) to achieve the same thing and have pagination support.

**Reply by Rick45** [reply]
Thanks for the help, I'm going to give it a try.

**Reply by projek01** [reply]
Hi Rick45, do you have a script that works beyond 1000 assets? Thank you in advance

**Reply by Rick45** [reply]
yes, let me publish it. 
i have build a small python script that runs on a docker to do this "smart" albuns :P

**Comment by Rick45** [comment]
@projek01  i have pushed it here, lets see if at EDO i add a read me with instructions

[https://github.com/Rick45/immich_autoAlbum](https://github.com/Rick45/immich_autoAlbum)

**Reply by projek01** [reply]
that is awesome! thank you very much for doing that.

**Reply by Rick45** [reply]
Added the documentation to it.
